### PR TITLE
Docs: more accurate yoda-speak

### PR DESCRIPTION
**Pull request by tonylukasavage**

### CODE REVIEW

**Comment by platinumazure** [comment]
I think the rule is only concerned about the objects-before-subjects part of Yoda-speak, not the nouns-before-verbs part.

On the other hand, for non-English speakers and/or non-Star Wars fans, the change will seem confusing or possibly incorrect.

On yet another hand, "red the color is" suggest postfix notation ("red" color ===), which JavaScript doesn't support.

Overall I'm :-1: on this change- too much confusion introduced, in exchange for authenticity to a dialect that was never really intended.

**Comment by pedrottimark** [comment]
@platinumazure Agree with your analysis of the first attempt at a change.

This came up recently in #6252 (which was not merged) so there is some evidence in favor of a change that removes distraction for readers who know Star Wars without adding confusion for readers who do not.

@tonylukasavage Can you instead suggest a rewording of the sentence to acknowledge that a so-called Yoda condition is only a loose analogy.

https://en.wikipedia.org/wiki/Yoda_conditions has an example: “This order is comparable to the non-standard speaking style of Yoda”

P.S. This is a negative example of a principle in technical training that a misleading analogy is worse that no analogy. For example, two strikes and you’re out.

**Comment by pedrottimark** [comment]
@tonylukasavage Forgot to mention can you please change the commit message to begin with **Docs:** instead of **Update:** for more information, see http://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changesa-namestep2a

**Comment by tonylukasavage** [comment]
Fellas, I just posted this up to have a little fun while me and the crew were going through our ruleset. It wasn't genuinely confusing for me, though I do agree the analogy is slightly askew. The wikipedia wording seems fair.

**Comment by pedrottimark** [comment]
😀 Sorry to seem too serious. If you can read that paragraph and adapt (not copy, please) to this context, we would welcome your contribution. We are up to 2 strikes, and I am happy if we can solve the issue before the count reaches 3.

File: docs/rules/yoda.md [hunk 1]
```diff
@@ -8,15 +8,15 @@ if ("red" === color) {
 }
 ```
 
-This is called a Yoda condition because it reads as, "red is the color", similar to the way the Star Wars character Yoda speaks. Compare to the other way of arranging the operands:
+This is called a Yoda condition because it reads as, "if red equals the color...", similar to the way the Star Wars character Yoda speaks. Compare to the other way of arranging the operands:
```
Comment (nzakas) [inline docs/rules/yoda.md]: "..."  isn't necessary

File: docs/rules/yoda.md [hunk 2]
```diff
@@ -8,15 +8,15 @@ if ("red" === color) {
...
 
 ```js
 if (color === "red") {
     // ...
 }
 ```
 
-This typically reads, "color is red", which is arguably a more natural way to describe the comparison.
+This typically reads, "if the color equals red...", which is arguably a more natural way to describe the comparison.
```
Comment (nzakas) [inline docs/rules/yoda.md]: Same hers

**Comment by nzakas** [comment]
Good enough, time to move on. :)

**Pull request merged**